### DISCUSSION TITLE
plugins are not working

### DISCUSSION THREAD

**Discussion by DartMitai** [jarun/nnn#1798] [Q&A]
Arch + Hyprland + Alacritty I can't open the image(imgview) or run (preview-tabbed) installed imv, zathura, xdotool, sxiv
![image](https://github.com/jarun/nnn/assets/104931116/599b3bae-d2bb-46eb-8a1e-df7236d0963c)

**Comment by manceras** [comment]
Are the variables correctly exported, try `echo $NNN_PLUG` in a terminal. If not, have you sourced that file in `config.fish`?

**Reply by DartMitai** [reply]
![изображение](https://github.com/jarun/nnn/assets/104931116/f717179e-3725-4148-bec5-7012637533db)

**Reply by manceras** [reply]
Are the plugins files you're trying to use under `~/.config/nnn/plugins`?

**Reply by DartMitai** [reply]
yep
![image](https://github.com/jarun/nnn/assets/104931116/e40c139d-4341-4f45-917e-409bdfc59a38)

**Comment by fuguesoft** [comment]
I just had an issue with plugins in fish shell. What I ended up doing was this:
```sh
﻿# ~/.config/fish/functions/n.fish
﻿function n --wraps=nnn --wraps='nnn -e' --description 'support nnn quit and change directory'
    export NNN_FIFO=/tmp/nnn.fifo nnn
    export NNN_PLUG='f:finder;o:fzopen;b:bulknew;p:preview-tui;l:launch;t:-!|tree -ps;d:diffs;n:nbak;x:fzplug' 
    # Block nesting of nnn in subshells
    if test -n "$NNNLVL" -a "$NNNLVL" -ge 1
        echo "nnn is already running"
        return
    end
    
    # The behaviour is set to cd on quit (nnn checks if NNN_TMPFILE is set)
    # If NNN_TMPFILE is set to a custom path, it must be exported for nnn to
    # see. To cd on quit only on ^G, remove the "-x" from both lines below,
    # without changing the paths.
    if test -n "$XDG_CONFIG_HOME"
        set -x NNN_TMPFILE "$XDG_CONFIG_HOME/nnn/.lastd"
    else
        set -x NNN_TMPFILE "$HOME/.config/nnn/.lastd"
    end

    # Unmask ^Q (, ^V etc.) (if required, see `stty -a`) to Quit nnn
    # stty start undef
    # stty stop undef
    # stty lwrap undef
    # stty lnext undef

    # The command function allows one to alias this function to `nnn` without
    # making an infinitely recursive alias
    command nnn -ae $argv

    if test -e $NNN_TMPFILE
        source $NNN_TMPFILE
        rm -- $NNN_TMPFILE
    end
end
```

**Reply by fuguesoft** [reply]
I wrapped my nnn in an alias with `alias -s n="nnn <flags_go_here>"`
Is it possible the nnn.fish file is conflicting with the nnn executable in `.local/bin` or where ever in your path?
If you hit the `;` key, do you see the `i` and `t` options listed in the status bar?

Por ejemplo:

https://github.com/user-attachments/assets/7b0d672f-0f79-4c3d-8120-3c7888d92038




[oh-snap.webm](https://github.com/user-attachments/assets/3b595dbd-dce8-489f-bdd1-ae38192aeade)

**Reply by fuguesoft** [reply]
Sorry about the two videos, don't think my current screen recorder makes web compatible vids in the first go
